### PR TITLE
Add vm support to NetBoxInventory

### PR DESCRIPTION
**Pull request by patrickdaj**
Ok heres a new PR.  Kinda got a bit screwed up with the rebase.  This should address all your comments.  Fixed the _get_resource function to _get_resources, dropped that extra function and addressed the docstring.

### CODE REVIEW

File: nornir_netbox/plugins/inventory/netbox.py [hunk 1]
```diff
@@ -263,3 +267,23 @@ def load(self) -> Inventory:
...
+
+    def _get_resources(self, url: str, params: Dict[str, Any]) -> List[Dict[str, Any]]:
+
+        resources: List[Dict[str, Any]] = []
+
+        while url:
+            r = self.session.get(url, params=params)
+
+            if not r.status_code == 200:
+                print(r)
```
Comment (wvandeun) [inline nornir_netbox/plugins/inventory/netbox.py:279]: remove the print statement

File: tests/test_netbox.py [hunk 2]
```diff
@@ -29,7 +29,29 @@ def get_inv(requests_mock, plugin, pagination, version, **kwargs):
                     json=json.load(f),
                     headers={"Content-type": "application/json"},
                 )
-    return plugin().load()
+    return plugin(include_vms=include_vms).load()
+
+
+def get_vm_inv(requests_mock, plugin, pagination, version, **kwargs):
```
Comment (wvandeun) [inline tests/test_netbox.py:35]: This function basically duplicates the code in get_inv().
Maybe we need to extract setting up the mocks into a different function and  call it from get_inv(). 

For example
```
def _create_mock(requests.mock, pagination, version, application, resource, **kwargs):
    """initialises mock objects for testcase"""
    if not pagination:
        with open(f"{BASE_PATH}/mocked/{version}/{resource}.json", "r") as f:
            requests_mock.get(
                "http://localhost:8080/api/{application}/{resource}/?limit=0",
                json=json.load(f),
                headers={"Content-type": "application/json"},
            )
    else:
        for offset in range(3):
            with open(f"{BASE_PATH}/mocked/{version}/{resource}-{offset}.json", "r") as f:
                url = "http://localhost:8080/api/{application}/{resource}/?limit=0"
                requests_mock.get(
                    f"{url}&offset={offset}" if offset else url,
                    json=json.load(f),
                    headers={"Content-type": "application/json"}
                )
```
```
def get_inv(requests_mock, plugin, pagination, version, include_vms=False, **kwargs):

    _create_mock(requests.mock, pagination, version, "dcim", "devices")
    if include_vms:
        _create_mock(requests_mock, pagination, version, "virtualization", "virtual-machines")

    return plugin(include_vms=include_vms).load()
```

**Review by wvandeun — CHANGES_REQUESTED** [review]

File: tests/test_netbox.py (hunk 2 above)
Comment (wvandeun) [reply tests/test_netbox.py:35]: probably also means you will have to rename the data mock files to "virtual-machines".

File: nornir_netbox/plugins/inventory/netbox.py (hunk 1 above)
Comment (patrickdaj) [reply nornir_netbox/plugins/inventory/netbox.py]: Done.

File: tests/test_netbox.py (hunk 2 above)
Comment (patrickdaj) [reply tests/test_netbox.py]: Dropped that in and was working with just a few tweaks.

File: tests/test_netbox.py [hunk 3]
```diff
@@ -12,46 +12,37 @@
...
-                requests_mock.get(
-                    f"{url}&offset={offset}" if offset else url,
-                    json=json.load(f),
-                    headers={"Content-type": "application/json"},
-                )
-    return get_inv(requests_mock, plugin, pagination, version, include_vms=True)
+def get_inv(requests_mock, plugin, pagination, version, **kwargs):
+
+    _create_mock(requests_mock, pagination, version, "dcim", "devices")
+    if "include_vms" in kwargs and kwargs["include_vms"]:
```
Comment (wvandeun) [inline tests/test_netbox.py:40]: `kwargs.get("include_vms", False)`

File: tests/test_netbox.py [hunk 4]
```diff
@@ -82,7 +73,7 @@ class TestNetBoxInventory2(TestNBInventory):
     # only on NetBoxInventory2 and NetBox 2.8.9
     @pytest.mark.parametrize("version", ["2.8.9"])
     def test_vms_inventory(self, requests_mock, version):
```
Comment (wvandeun) [inline tests/test_netbox.py:75]: Would suggest to rename this test to `test_inventory_include_vms` as is better covers what it actually does.
The current name could be interpreted as if it is only testing VM's inventory, which is not the case.

File: tests/test_netbox.py [hunk 5]
```diff
@@ -91,7 +82,7 @@ def test_vms_inventory(self, requests_mock, version):
 
     @pytest.mark.parametrize("version", ["2.8.9"])
     def test_vms_inventory_pagination(self, requests_mock, version):
```
Comment (wvandeun) [inline tests/test_netbox.py:84]: Would also rename for the same reason.
`test_inventory_include_vms_pagination` ?

**Review by wvandeun — CHANGES_REQUESTED** [review]

**Comment by wvandeun** [comment]
Starting to look good, almost there :)

**Comment by patrickdaj** [comment]
> Starting to look good, almost there :)

Thanks for the detailed review.

File: tests/test_netbox.py (hunk 5 above)
Comment (patrickdaj) [reply tests/test_netbox.py]: Done.

File: tests/test_netbox.py (hunk 4 above)
Comment (patrickdaj) [reply tests/test_netbox.py]: Done.  Makes sense.

File: tests/test_netbox.py (hunk 3 above)
Comment (patrickdaj) [reply tests/test_netbox.py]: Meant to do that.  :)  Still hanging on to some old methods.

**Comment by wvandeun** [comment]
LGTM, can you squash your commits and add a edit the commit message? "implements support for virtual-machines in NetBoxInventory2" 
Thx!

**Comment by patrickdaj** [comment]
> LGTM, can you squash your commits and add a edit the commit message? "implements support for virtual-machines in NetBoxInventory2"
> Thx!

Not 100% on how to do this.  Thought you could do that via github in the PR.  Anyhow, I ran 'git rebase -i HEAD~3', set last two commits to squash and set the new commit message.  Problem is it won't let me push without pulling first which creates a new merge commit and thus makes me fail to squash down to a single commit message.  Any help appreciated.

**Comment by wvandeun** [comment]
The problem with doing it in the Github interface is that we don't have a merge commit anymore. 

You should indeed do like you are mentioning, but you will need to force push. `git push -f origin`

**Review by wvandeun — APPROVED** [review]

**Comment by patrickdaj** [comment]
> The problem with doing it in the Github interface is that we don't have a merge commit anymore.
> 
> You should indeed do like you are mentioning, but you will need to force push. `git push -f origin`
Thanks.  Looks like that worked.

**Comment by wvandeun** [comment]
Thx for your work on this

**Pull request merged**